cmd/cerbosctl/put/schema.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2021-2022 Zenauth Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+package put
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"os"
+
+	"github.com/alecthomas/kong"
+
+	"github.com/cerbos/cerbos/client"
+	cmdclient "github.com/cerbos/cerbos/cmd/cerbosctl/internal/client"
+	"github.com/cerbos/cerbos/internal/util"
+)
+
+const schemaCmdHelp = `# Put schemas
+cerbosctl put schemas ./path/to/schema.json
+cerbosctl put schema ./path/to/schema.json
+cerbosctl put s ./path/to/schema.json
+
+# Put multiple schemas
+cerbosctl put schema ./path/to/schema.json ./path/to/other/schema.json
+
+# Put schemas under a directory
+cerbosctl put schema ./dir/to/schemas ./other/dir/to/schemas
+
+# Put schemas under a directory recursively
+cerbosctl put schema --recursive ./dir/to/schemas
+cerbosctl put schema -R ./dir/to/schemas`
+
+type SchemaCmd struct {
+	Paths []string `arg:"" type:"path" help:"Path to schema file or directory"`
+}
+
+func (sc *SchemaCmd) Run(k *kong.Kong, put *Cmd, ctx *cmdclient.Context) error {
+	if len(sc.Paths) == 0 {
+		return fmt.Errorf("no filename(s) provided")
+	}
+
+	schemas, err := sc.findFiles(sc.Paths, put.Recursive)
+	if err != nil {
+		return err
+	}
+
+	err = ctx.AdminClient.AddOrUpdateSchema(context.TODO(), schemas)
+	if err != nil {
+		return fmt.Errorf("failed to add or update the schemas: %w", err)
+	}
+
+	return nil
+}
+
+func (sc *SchemaCmd) Help() string {
+	return schemaCmdHelp
+}
+
+func (sc *SchemaCmd) findFiles(paths []string, recursive bool) (*client.SchemaSet, error) {
+	schemas := client.NewSchemaSet()
+	for _, path := range paths {
```

Comment:
A lot of this code is the same as the code for the put policy call. I think we can make it into a single function that invokes a callback for each file found. The callback function can take care of the special policy or schema handling code. That way we don't have to maintain two copies of the traversal code.

```go
func findFiles(paths []string, recursive bool, callback func(filePath string) error) error 
```